routes/auth.js

```diff
@@ -0,0 +1,32 @@
+const bcrypt = require('bcrypt');
+const express= require('express');
+const router = express.Router();
+const mongoose = require('mongoose');
+const Joi = require('joi');
+const {User, validate} = require('../models/user');
+
+router.post('/', async (req,res)=>{             //This is for logins
+    const {error} = validate(req.body);        // This validates the data put in and doesn't allow an invalid type
+    if (error) return res.status(400).send(error.details[0].message);
+
+    let user = await User.findOne({email: req.body.email});     //This function checks whether the email is the same as the one in database
+    if (!user) return res.status(400).send('Nope! The password or email is invalid - try again!');
```

Comment:
Great job, detailed enough to even have proper status codes :+1: 
However this is a RESTful JSON application and you are returning a string. Try returning an object, eg.

```
{
   status: 'ok',
   message: 'Nope! The password or email is invalid - try again!'
}
```

This applies here and everywhere where you return a string